denseclus/DenseClus.py

```diff
@@ -34,6 +34,22 @@
 import umap.umap_ as umap
 from sklearn.base import BaseEstimator, ClassifierMixin
 
+
+# CUML UMAP and HDBSCAN integration inspired by and adapted from https://github.com/ddangelov/Top2Vec/tree/master
```

Comment:
@bharven Instead of try/except can you try something with importlib or pkgutil? For importlib it would be something like:
 ```import importlib.util

_HAVE_CUML = importlib.util.find_spec('cuml') is not None
_HAVE_CUHDBSCAN = importlib.util.find_spec('cuml.cluster.HDBSCAN') is not None
```